### DISCUSSION TITLE
List this account on the directory - default setting on Mastodon server

### DISCUSSION THREAD

**Discussion by vserbu** [mastodon/mastodon#17218] [Q&A]
Hi.

I'm administering a small Mastodon instance and I would like to make users to opt-in their accounts on the directory. Is there any way the option under Profile - Appearance - List this account on the directory to be selected by default. I know it is already "reccomended" but it is not selected by default, and new users usually leave everything by default, not tinkering around.

**Comment by vserbu** [comment]
I can't believe that there is no way to define that option as selected by default?

**Reply by ClearlyClaire** [reply]
There is no option to make it opt-out, it's opt-in by design.

**Reply by vserbu** [reply]
But it is not. Every user that registered on salocha.online is opt-out and I myself, when tried to register as another user is by defalut opt-out but it says it is recommended for "List this account on the directory".  Strange.

**Reply by ClearlyClaire** [reply]
“Opt-in” means you have to opt in to use the feature. It means it's off by default. It's off by default because we don't want people to expose themselves more than they would expect. It's “recommended” because it's nevertheless an important feature for discovery and would be on by default if it weren't for the privacy/visibility concerns.

**Reply by Gargron** [reply]
I think the profile *directory* might be on its way out - we have already renamed the setting to be more general - but I do agree that discoverability on a *social* network being opt-in is somewhat counter-intuitive...

**Comment by nemobis** [comment]
Thank you for opening this discussion! I somehow assumed it was enabled by default and wondered why nobody showed up in `/explore`...

It's also annoying that this preference doesn't seem to be controllable from the API. If you're using [mechanize](https://mechanize.readthedocs.io/en/latest/), you can do something like this:

```
    br = mechanize.Browser()
    br.open(f"{domain}/auth/sign_in")
    br.select_form(id="new_user")
    br.set_value(email, id="user_email")
    br.set_value(password, id="user_password")
    sleep(10)
    br.submit()
    br.open(f"{domain}/settings/profile")
    br.select_form(id="edit_profile")
    br.toggle_single(name="account[discoverable]")
    sleep(5)
    br.submit()
```
